service-base/src/main/java/org/eclipse/hono/service/resourcelimits/ResourceLimitChecks.java

```diff
@@ -44,6 +62,24 @@
      *         <p>
      *         The future will be failed with a {@link ServiceInvocationException}
      *         if the check could not be performed.
+     * @deprecated Use {@link #isMessageLimitReached(TenantObject, long, SpanContext)} instead.
      */
+    @Deprecated
     Future<Boolean> isMessageLimitReached(TenantObject tenantObject, long payloadSize);
+
+    /**
+     * Checks if the maximum limit for the messages configured for a tenant
+     * have been reached.
+     *
+     * @param tenantObject The tenant configuration to check the limit against.
+     * @param payloadSize The message payload size in bytes.
+     * @param spanContext The currently active OpenTracing span context that is used to
+     *                    trace the limits verification or {@code null}
+     *                    if no span is currently active.
+     * @return A future indicating the outcome of the check.
+     *         <p>
+     *         The future will be failed with a {@link ServiceInvocationException}
+     *         if the check could not be performed.
+     */
+    Future<Boolean> isMessageLimitReached(TenantObject tenantObject, long payloadSize, SpanContext spanContext);
```

Comment:
what if tenant or span context are `null`?

Reply:
javadoc has been appended with ` @throws NullPointerException if the tenant object is null.`